File: archive/canvas-api/src/test/java/CalenderEventUnitTest.java

/*
 * Copyright (C) 2017 - present Instructure, Inc.
 *
 *     Licensed under the Apache License, Version 2.0 (the "License");
 *     you may not use this file except in compliance with the License.
 *     You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 *     Unless required by applicable law or agreed to in writing, software
 *     distributed under the License is distributed on an "AS IS" BASIS,
 *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *     See the License for the specific language governing permissions and
 *     limitations under the License.
 *
 */

import com.google.gson.Gson;
import com.instructure.canvasapi.model.ScheduleItem;
import com.instructure.canvasapi.utilities.CanvasRestAdapter;

import junit.framework.Assert;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.annotation.Config;

@Config(sdk = 17)
@RunWith(RobolectricGradleTestRunner.class)
public class CalenderEventUnitTest extends Assert {

    @Test
    public void testCalenderEvent(){
        Gson gson = CanvasRestAdapter.getGSONParser();
        ScheduleItem scheduleItem = gson.fromJson(calenderEventJSON, ScheduleItem.class);

        assertNotNull(scheduleItem);

        if(scheduleItem.isAllDay()){
            assertNotNull(scheduleItem.getAllDayDate());
        }else{
            assertFalse(scheduleItem.isAllDay());
            assertNull(scheduleItem.getAllDayDate());
        }

        assertTrue(scheduleItem.getId() > 0);
        assertNotNull(scheduleItem.getStartDate());
        assertNotNull(scheduleItem.getEndDate());
        assertNotNull(scheduleItem.getLocationName());
        assertNotNull(scheduleItem.getTitle());
        assertNotNull(scheduleItem.getDescription());
        assertNotNull(scheduleItem.getLocationAddress());
    }

    @Test
    public void testCalenderEventList(){
        Gson gson = CanvasRestAdapter.getGSONParser();
        ScheduleItem[] list = gson.fromJson(upcomingEventsJSON, ScheduleItem[].class);

        assertNotNull(list);

        ScheduleItem listScheduleItem = list[0];

        if(listScheduleItem.isAllDay()){
            assertNotNull(listScheduleItem.getAllDayDate());
        }else{
            assertFalse(listScheduleItem.isAllDay());
            assertNull(listScheduleItem.getAllDayDate());
        }

        assertTrue(listScheduleItem.getId() > 0);
        assertNotNull(listScheduleItem.getStartDate());
        assertNotNull(listScheduleItem.getEndDate());
        assertNotNull(listScheduleItem.getLocationName());
        assertNotNull(listScheduleItem.getTitle());
        assertNotNull(listScheduleItem.getDescription());
        assertNotNull(listScheduleItem.getLocationAddress ());
    }




    //vanilla event request
    //@GET("/calendar_events/{event_id}")
    //void getCalendarEvent(@Path("event_id") long event_id, Callback<ScheduleItem> callback);
    final String calenderEventJSON = "{\n" +
            "\"all_day\": false,\n" +
            "\"all_day_date\": null,\n" +
            "\"created_at\": \"2014-07-15T21:18:24Z\",\n" +
            "\"end_at\": \"2014-07-17T00:00:00Z\",\n" +
            "\"id\": 1935799,\n" +
            "\"location_address\": \"Hodor hodor hodor\",\n" +
            "\"location_name\": \"Here\",\n" +
            "\"start_at\": \"2014-07-16T23:00:00Z\",\n" +
            "\"title\": \"Hodor's Reckoning\",\n" +
            "\"updated_at\": \"2014-07-15T22:03:20Z\",\n" +
            "\"workflow_state\": \"active\",\n" +
            "\"description\": \"<p>Hodorrrrrr</p>\",\n" +
            "\"context_code\": \"user_5814789\",\n" +
            "\"child_events_count\": 0,\n" +
            "\"parent_event_id\": null,\n" +
            "\"hidden\": false,\n" +
            "\"child_events\": [],\n" +
            "\"url\": \"https://mobiledev.instructure.com/api/v1/calendar_events/1935799\",\n" +
            "\"html_url\": \"https://mobiledev.instructure.com/calendar?event_id=1935799&include_contexts=user_5814789#7b2273686f77223a2267726f75705f757365725f35383134373839227d\"\n" +
            "}";

    //upcoming events request
    //@GET("/users/self/upcoming_events")
    //ScheduleItem[] getUpcomingEvents();
    final String upcomingEventsJSON = "[\n" +
            "{\n" +
            "\"all_day\": false,\n" +
            "\"all_day_date\": null,\n" +
            "\"created_at\": \"2014-07-15T21:18:24Z\",\n" +
            "\"end_at\": \"2014-07-17T00:00:00Z\",\n" +
            "\"id\": 1935799,\n" +
            "\"location_address\": \"Hodor hodor hodor\",\n" +
            "\"location_name\": \"Here\",\n" +
            "\"start_at\": \"2014-07-16T23:00:00Z\",\n" +
            "\"title\": \"Hodor's Reckoning\",\n" +
            "\"updated_at\": \"2014-07-15T22:03:20Z\",\n" +
            "\"workflow_state\": \"active\",\n" +
            "\"description\": \"<p>Hodorrrrrr</p>\",\n" +
            "\"context_code\": \"user_5814789\",\n" +
            "\"child_events_count\": 0,\n" +
            "\"parent_event_id\": null,\n" +
            "\"hidden\": false,\n" +
            "\"child_events\": [],\n" +
            "\"url\": \"https://mobiledev.instructure.com/api/v1/calendar_events/1935799\",\n" +
            "\"html_url\": \"https://mobiledev.instructure.com/calendar?event_id=1935799&include_contexts=user_5814789#7b2273686f77223a2267726f75705f757365725f35383134373839227d\"\n" +
            "},\n" +
            "{\n" +
            "\"all_day\": false,\n" +
            "\"all_day_date\": null,\n" +
            "\"created_at\": \"2014-07-15T22:26:03Z\",\n" +
            "\"end_at\": \"2014-07-18T00:00:00Z\",\n" +
            "\"id\": 1935841,\n" +
            "\"location_address\": null,\n" +
            "\"location_name\": \"lol\",\n" +
            "\"start_at\": \"2014-07-17T22:00:00Z\",\n" +
            "\"title\": \"Woop\",\n" +
            "\"updated_at\": \"2014-07-15T22:26:03Z\",\n" +
            "\"workflow_state\": \"active\",\n" +
            "\"description\": null,\n" +
            "\"context_code\": \"user_5814789\",\n" +
            "\"child_events_count\": 0,\n" +
            "\"parent_event_id\": null,\n" +
            "\"hidden\": false,\n" +
            "\"child_events\": [],\n" +
            "\"url\": \"https://mobiledev.instructure.com/api/v1/calendar_events/1935841\",\n" +
            "\"html_url\": \"https://mobiledev.instructure.com/calendar?event_id=1935841&include_contexts=user_5814789#7b2273686f77223a2267726f75705f757365725f35383134373839227d\"\n" +
            "},\n" +
            "{\n" +
            "\"all_day\": false,\n" +
            "\"all_day_date\": null,\n" +
            "\"created_at\": \"2014-07-15T22:26:25Z\",\n" +
            "\"end_at\": \"2014-07-18T21:00:00Z\",\n" +
            "\"id\": 1935842,\n" +
            "\"location_address\": null,\n" +
            "\"location_name\": \"lololol\",\n" +
            "\"start_at\": \"2014-07-18T19:00:00Z\",\n" +
            "\"title\": \"Poow\",\n" +
            "\"updated_at\": \"2014-07-15T22:26:25Z\",\n" +
            "\"workflow_state\": \"active\",\n" +
            "\"description\": null,\n" +
            "\"context_code\": \"user_5814789\",\n" +
            "\"child_events_count\": 0,\n" +
            "\"parent_event_id\": null,\n" +
            "\"hidden\": false,\n" +
            "\"child_events\": [],\n" +
            "\"url\": \"https://mobiledev.instructure.com/api/v1/calendar_events/1935842\",\n" +
            "\"html_url\": \"https://mobiledev.instructure.com/calendar?event_id=1935842&include_contexts=user_5814789#7b2273686f77223a2267726f75705f757365725f35383134373839227d\"\n" +
            "}\n" +
            "]";
}

File: libs/canvas-api-2/src/test/java/com/instructure/canvasapi2/unit/CalenderEventUnitTest.kt
/*
 * Copyright (C) 2017 - present Instructure, Inc.
 *
 *     Licensed under the Apache License, Version 2.0 (the "License");
 *     you may not use this file except in compliance with the License.
 *     You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 *     Unless required by applicable law or agreed to in writing, software
 *     distributed under the License is distributed on an "AS IS" BASIS,
 *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *     See the License for the specific language governing permissions and
 *     limitations under the License.
 *
 */

package com.instructure.canvasapi2.unit

import com.instructure.canvasapi2.models.ScheduleItem
import com.instructure.canvasapi2.utils.parse
import org.junit.Assert
import org.intellij.lang.annotations.Language
import org.junit.Test

class CalenderEventUnitTest : Assert() {

    @Test
    fun testCalenderEvent() {
        val scheduleItem: ScheduleItem = calenderEventJSON.parse()

        Assert.assertNotNull(scheduleItem)

        if (scheduleItem.isAllDay) {
            Assert.assertNotNull(scheduleItem.allDayAt)
        } else {
            Assert.assertFalse(scheduleItem.isAllDay)
            Assert.assertNull(scheduleItem.allDayAt)
        }

        Assert.assertTrue(scheduleItem.id > 0)
        Assert.assertNotNull(scheduleItem.startAt)
        Assert.assertNotNull(scheduleItem.endAt)
        Assert.assertNotNull(scheduleItem.locationName)
        Assert.assertNotNull(scheduleItem.title)
        Assert.assertNotNull(scheduleItem.description)
        Assert.assertNotNull(scheduleItem.locationAddress)
    }

    @Test
    fun testCalenderEventList() {
        val list: Array<ScheduleItem> = upcomingEventsJSON.parse()

        Assert.assertNotNull(list)

        val listScheduleItem = list[0]

        if (listScheduleItem.isAllDay) {
            Assert.assertNotNull(listScheduleItem.allDayAt)
        } else {
            Assert.assertFalse(listScheduleItem.isAllDay)
            Assert.assertNull(listScheduleItem.allDayAt)
        }

        Assert.assertTrue(listScheduleItem.id > 0)
        Assert.assertNotNull(listScheduleItem.startAt)
        Assert.assertNotNull(listScheduleItem.endAt)
        Assert.assertNotNull(listScheduleItem.locationName)
        Assert.assertNotNull(listScheduleItem.title)
        Assert.assertNotNull(listScheduleItem.description)
        Assert.assertNotNull(listScheduleItem.locationAddress)
    }

    /**
     * vanilla event request
     * @GET("/calendar_events/{event_id}")
     * void getCalendarEvent(@Path("event_id") long event_id, Callback<ScheduleItem> callback);
     */
    @Language("JSON")
    private val calenderEventJSON = """
      {
        "all_day": false,
        "all_day_date": null,
        "created_at": "2014-07-15T21:18:24Z",
        "end_at": "2014-07-17T00:00:00Z",
        "id": 1935799,
        "location_address": "Hodor hodor hodor",
        "location_name": "Here",
        "start_at": "2014-07-16T23:00:00Z",
        "title": "Hodor's Reckoning",
        "updated_at": "2014-07-15T22:03:20Z",
        "workflow_state": "active",
        "description": "<p>Hodorrrrrr</p>",
        "context_code": "user_5814789",
        "child_events_count": 0,
        "parent_event_id": null,
        "hidden": false,
        "child_events": [],
        "url": "https://mobiledev.instructure.com/api/v1/calendar_events/1935799",
        "html_url": "https://mobiledev.instructure.com/calendar?event_id=1935799&include_contexts=user_5814789#7b2273686f77223a2267726f75705f757365725f35383134373839227d"
      }"""

    /**
     * upcoming events request
     * @GET("/users/self/upcoming_events")
     * ScheduleItem[] getUpcomingEvents();
     */
    @Language("JSON")
    private val upcomingEventsJSON = """
      [
        {
          "all_day": false,
          "all_day_date": null,
          "created_at": "2014-07-15T21:18:24Z",
          "end_at": "2014-07-17T00:00:00Z",
          "id": 1935799,
          "location_address": "Hodor hodor hodor",
          "location_name": "Here",
          "start_at": "2014-07-16T23:00:00Z",
          "title": "Hodor's Reckoning",
          "updated_at": "2014-07-15T22:03:20Z",
          "workflow_state": "active",
          "description": "<p>Hodorrrrrr</p>",
          "context_code": "user_5814789",
          "child_events_count": 0,
          "parent_event_id": null,
          "hidden": false,
          "child_events": [],
          "url": "https://mobiledev.instructure.com/api/v1/calendar_events/1935799",
          "html_url": "https://mobiledev.instructure.com/calendar?event_id=1935799&include_contexts=user_5814789#7b2273686f77223a2267726f75705f757365725f35383134373839227d"
        },
        {
          "all_day": false,
          "all_day_date": null,
          "created_at": "2014-07-15T22:26:03Z",
          "end_at": "2014-07-18T00:00:00Z",
          "id": 1935841,
          "location_address": null,
          "location_name": "lol",
          "start_at": "2014-07-17T22:00:00Z",
          "title": "Woop",
          "updated_at": "2014-07-15T22:26:03Z",
          "workflow_state": "active",
          "description": null,
          "context_code": "user_5814789",
          "child_events_count": 0,
          "parent_event_id": null,
          "hidden": false,
          "child_events": [],
          "url": "https://mobiledev.instructure.com/api/v1/calendar_events/1935841",
          "html_url": "https://mobiledev.instructure.com/calendar?event_id=1935841&include_contexts=user_5814789#7b2273686f77223a2267726f75705f757365725f35383134373839227d"
        },
        {
          "all_day": false,
          "all_day_date": null,
          "created_at": "2014-07-15T22:26:25Z",
          "end_at": "2014-07-18T21:00:00Z",
          "id": 1935842,
          "location_address": null,
          "location_name": "lololol",
          "start_at": "2014-07-18T19:00:00Z",
          "title": "Poow",
          "updated_at": "2014-07-15T22:26:25Z",
          "workflow_state": "active",
          "description": null,
          "context_code": "user_5814789",
          "child_events_count": 0,
          "parent_event_id": null,
          "hidden": false,
          "child_events": [],
          "url": "https://mobiledev.instructure.com/api/v1/calendar_events/1935842",
          "html_url": "https://mobiledev.instructure.com/calendar?event_id=1935842&include_contexts=user_5814789#7b2273686f77223a2267726f75705f757365725f35383134373839227d"
        }
      ]"""
}
